modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/impl/IngestServiceImpl.java

```diff
@@ -918,15 +922,38 @@ public MediaPackage addCatalog(InputStream in, String fileName, MediaPackageElem
    */
   @Override
   public MediaPackage addCatalog(InputStream in, String fileName, MediaPackageElementFlavor flavor, String[] tags,
-          MediaPackage mediaPackage) throws IOException, IngestException {
+          MediaPackage mediaPackage) throws IOException, IngestException, IllegalArgumentException {
     Job job = null;
     try {
       job = serviceRegistry.createJob(JOB_TYPE, INGEST_CATALOG, null, null, false, ingestFileJobLoad);
       job.setStatus(Status.RUNNING);
       job = serviceRegistry.updateJob(job);
-      String elementId = UUID.randomUUID().toString();
-      logger.info("Start adding catalog {} from input stream on mediapackage {}", elementId, mediaPackage);
-      URI newUrl = addContentToRepo(mediaPackage, elementId, fileName, in);
+      final String elementId = UUID.randomUUID().toString();
+      final String mediaPackageId = mediaPackage.getIdentifier().toString();
+      logger.info("Start adding catalog {} from input stream on mediapackage {}", elementId, mediaPackageId);
+      final URI newUrl = addContentToRepo(mediaPackage, elementId, fileName, in);
+
+      final boolean isJSON;
+      try (InputStream inputStream = workingFileRepository.get(mediaPackageId, elementId)) {
+        try (BufferedReader reader  = new BufferedReader(new InputStreamReader(inputStream, StandardCharsets.UTF_8))) {
+          // Exception for current BBB integration which is ingesting a JSON object as catalog
+          isJSON = reader.read() == '{';
```

Comment:
This fails for JsonArrays, which are expected by [Cut Marks To Smil](https://docs.opencast.org/r/10.x/admin/#workflowoperationhandlers/cut-marks-to-smil-woh.html).

Reply:
Ups… I was sure that was an object.
But then let's change this to

```java
isJSON = reader.read() == '[';
```

Reply:
Fixed